runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -167,20 +182,15 @@ public void write(IToken token, Object data) throws OverwriteException {
      * @return A result, which be cached.
      */
     public Map<Long, ILogData> read(Iterable<Long> addresses) {
-        Map<Long, ILogData> addressesMap;
-        if (!runtime.isCacheDisabled()) {
-            addressesMap = readCache.getAll(addresses);
-        } else {
-            addressesMap = this.cacheFetch(addresses);
-        }
+        return read(addresses, null);
+    }
 
-        for (ILogData logData : addressesMap.values()) {
-            if (logData.isTrimmed()) {
-                throw new TrimmedException();
-            }
+    public Map<Long, ILogData> read(Iterable<Long> addresses, AddressSpaceOptions opts) {
+        if ((opts == null || opts != null && opts.isDoCache() )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)